### ISSUE TITLE
Make it possible to use the original format data

### ISSUE THREAD

**Issue by johann-petrak** [GateNLP/gate-lf-pytorch-json#45]
Make it possible to use the original format data so that the model can make use of the actual text and use character embeddings or similar. 
This will be based on Xingyi's code from https://github.com/GateNLP/AbuseDetection
Created branch `1905useorig` for this.

**Comment by johann-petrak** [comment]
TODO: initially there will be a dependency on allennlp for use with the elmo models, figure out a way to remove that dependency. Either make the modelwrapperdefault only depend on this if an elmo model is actually used or have a look to see if we can just merge or reimplement the code needed. That way, only an actual ELMO-based model would depend on that library

**Comment by johann-petrak** [comment]
Probably best to delegate this to the module somehow: if we have the config parameters elmo or orig set it means that we use a module that needs the original data. In that case we will expect the module to have the methods needed in the the default wrapper in `_apply_model`

**Comment by johann-petrak** [comment]
The code by XS for training an elmo model uses a new method to split the validation set, but maybe we can make use of the batch generator code instead?

**Comment by johann-petrak** [comment]
After merging the fix for #34 the default doc classification results stay the same. 

Problem when texting the module TextClassCnnSingleElmo: no proper error handling if elmo model file is missing / not specified.

**Comment by johann-petrak** [comment]
TODO 1: once we know we do not need the the converted data, avoid generating it.
TODO 2: see of we can move the code to generate the target indices from the literal target strings into the module for all training situations

**Comment by johann-petrak** [comment]
TODO 2: for now we do this in the wrapper, since the proper way to move this into the module is to move the whole evaluation code (either inherit a proper default or override a specific way to do it, e.g. for CNNs)

**Comment by johann-petrak** [comment]
OK, TODO1 works: if we have elmo/orig, no conversion is done and therefore also no converted dataset kept around.

**Comment by johann-petrak** [comment]
Quick test of application: seems to work.
This can get merged now.
